contrib/babelfishpg_money/fixeddecimal.c

```
@@ -409,9 +409,10 @@ fixeddecimal2str(int64 val, char *buffer,
 	 * By default, fixeddecimal contains 4 digits after decimal but if we only need
 	 * 2 decimal points, remove last two decimal digits, round off remaining value.
 	 */
-	if (fixeddecimal_scale == 2)
+	if (fixeddecimal_scale < 4)
```

You should use existing macro to replace `4`. Otherwise you can raise an error, if the input argument is greater than 4

---

Fixeddecimal scale has maximum value 4 so it can't be higher. Moreover, this is an internal function where we have made sure to pass valid scale values, there is no way a user can directly interact with this function and supply invalid input values.